example/sign/lib/pages/w3m_page.dart

```diff
@@ -21,39 +20,34 @@ class W3MPage extends StatefulWidget {
 }
 
 class _W3MPageState extends State<W3MPage> with SingleTickerProviderStateMixin {
-  bool _initialized = false;
-
-  IW3MService? _w3mService;
-
+  late IW3MService _w3mService;
   bool _isConnected = false;
 
   @override
   void initState() {
     super.initState();
-
-    initialize();
-  }
-
-  Future<void> initialize() async {
-    _w3mService = W3MService(
-      web3App: widget.web3App,
-      recommendedWalletIds: {
-        'afbd95522f4041c71dd4f1a065f971fd32372865b416f95a0b1db759ae33f2a7',
-        '38f5d18bd8522c244bdd70cb4a68e0e718865155811c043f052fb9f1c51de662',
-        'c57ca95b47569778a828d19178114f4db188b89b763c899ba0be274e97267d96',
-      },
-    );
-
     widget.web3App.onSessionConnect.subscribe(_onWeb3AppConnect);
     widget.web3App.onSessionDelete.subscribe(_onWeb3AppDisconnect);
+    _initializeService();
+  }
 
-    await _w3mService?.init();
-
-    _isConnected = widget.web3App.sessions.getAll().isNotEmpty;
+  Future<bool> _initializeService() async {
+    try {
+      _w3mService = W3MService(
+        web3App: widget.web3App,
+        recommendedWalletIds: {
```

Comment:
Ideally these should be dynamic i.e. first 3 (or more) wallets returned from api

Reply:
`recommendedWalletIds` is used to give priority to wallets of choice. Check here https://github.com/WalletConnect/WalletConnectModalFlutter#excluding-recommending-and-including-wallets